src/python/pants/core/goals/lint.py

```diff
@@ -88,97 +92,124 @@ def prep_output(s: bytes) -> str:
             partition_description=partition_description,
             report=report,
         )
 
     def metadata(self) -> dict[str, Any]:
         return {"partition": self.partition_description}
 
-
-@frozen_after_init
-@dataclass(unsafe_hash=True)
-class LintResults(EngineAwareReturnType):
-    """Zero or more LintResult objects for a single linter.
-
-    Typically, linters will return one result. If they no-oped, they will return zero results.
-    However, some linters may need to partition their input and thus may need to return multiple
-    results. For example, many Python linters will need to group by interpreter compatibility.
-    """
-
-    results: tuple[LintResult, ...]
-    linter_name: str
-
-    def __init__(self, results: Iterable[LintResult], *, linter_name: str) -> None:
-        self.results = tuple(results)
-        self.linter_name = linter_name
-
-    @property
-    def skipped(self) -> bool:
-        return bool(self.results) is False
-
-    @memoized_property
-    def exit_code(self) -> int:
-        return next((result.exit_code for result in self.results if result.exit_code != 0), 0)
-
     def level(self) -> LogLevel | None:
-        if self.skipped:
-            return LogLevel.DEBUG
         return LogLevel.ERROR if self.exit_code != 0 else LogLevel.INFO
 
     def message(self) -> str | None:
-        if self.skipped:
-            return f"{self.linter_name} skipped."
         message = self.linter_name
         message += (
             " succeeded." if self.exit_code == 0 else f" failed (exit code {self.exit_code})."
         )
+        if self.partition_description:
+            message += f"\nPartition: {self.partition_description}"
+        if self.stdout:
+            message += f"\n{self.stdout}"
+        if self.stderr:
+            message += f"\n{self.stderr}"
+        if self.partition_description or self.stdout or self.stderr:
+            message += "\n\n"
 
-        def msg_for_result(result: LintResult) -> str:
-            msg = ""
-            if result.stdout:
-                msg += f"\n{result.stdout}"
-            if result.stderr:
-                msg += f"\n{result.stderr}"
-            if msg:
-                msg = f"{msg.rstrip()}\n\n"
-            return msg
-
-        if len(self.results) == 1:
-            results_msg = msg_for_result(self.results[0])
-        else:
-            results_msg = "\n"
-            for i, result in enumerate(self.results):
-                msg = f"Partition #{i + 1}"
-                msg += (
-                    f" - {result.partition_description}:" if result.partition_description else ":"
-                )
-                msg += msg_for_result(result) or "\n\n"
-                results_msg += msg
-        message += results_msg
         return message
 
     def cacheable(self) -> bool:
         """Is marked uncacheable to ensure that it always renders."""
         return False
 
 
+class LintPartitions(Collection[Tuple[Tuple[str, ...], Any]]):
```

Comment:
I wanna look into making this generic, so the user does `LintPartitions[MetadataTyoe]`.

I don't think the rule parser will like that though...